src/main/java/com/aws/greengrass/util/platforms/Platform.java

```diff
@@ -199,4 +198,13 @@ protected abstract FileSystemPermissionView getFileSystemPermissionView(FileSyst
     public abstract void setIpcFilePermissions(Path rootPath);
 
     public abstract void cleanupIpcFiles(Path rootPath);
+
+    public abstract void createResourceManager(String componentName);
```

Comment:
Instead of adding more and more into the platform space, I think we might want to split stuff out and add a generic `getPlatformResourceController` or something like that. And then that class can have all these methods. And of course it can be implemented for different platforms still.